From d9fe103dea50bdab6a2cf53dac820f0a25960395 Mon Sep 17 00:00:00 2001
From: Elijah Malaby <qwe12345678910@gmail.com>
Date: Sun, 12 Dec 2021 13:41:28 -0500
Subject: [PATCH 1/2] Add window divider transparency

---
 src/xterm.c | 44 ++++++++------------------------------------
 1 file changed, 8 insertions(+), 36 deletions(-)

diff --git a/src/xterm.c b/src/xterm.c
index d826eec2419..917185fac29 100644
--- a/src/xterm.c
+++ b/src/xterm.c
@@ -7469,42 +7469,14 @@ x_draw_window_divider (struct window *w, int x0, int x1, int y0, int y1)
 			      : FRAME_FOREGROUND_PIXEL (f));
   Display *display = FRAME_X_DISPLAY (f);
 
-  if ((y1 - y0 > x1 - x0) && (x1 - x0 >= 3))
-    /* A vertical divider, at least three pixels wide: Draw first and
-       last pixels differently.  */
-    {
-      XSetForeground (display, f->output_data.x->normal_gc, color_first);
-      x_fill_rectangle (f, f->output_data.x->normal_gc,
-			x0, y0, 1, y1 - y0, false);
-      XSetForeground (display, f->output_data.x->normal_gc, color);
-      x_fill_rectangle (f, f->output_data.x->normal_gc,
-			x0 + 1, y0, x1 - x0 - 2, y1 - y0, false);
-      XSetForeground (display, f->output_data.x->normal_gc, color_last);
-      x_fill_rectangle (f, f->output_data.x->normal_gc,
-			x1 - 1, y0, 1, y1 - y0, false);
-    }
-  else if ((x1 - x0 > y1 - y0) && (y1 - y0 >= 3))
-    /* A horizontal divider, at least three pixels high: Draw first and
-       last pixels differently.  */
-    {
-      XSetForeground (display, f->output_data.x->normal_gc, color_first);
-      x_fill_rectangle (f, f->output_data.x->normal_gc,
-			x0, y0, x1 - x0, 1, false);
-      XSetForeground (display, f->output_data.x->normal_gc, color);
-      x_fill_rectangle (f, f->output_data.x->normal_gc,
-			x0, y0 + 1, x1 - x0, y1 - y0 - 2, false);
-      XSetForeground (display, f->output_data.x->normal_gc, color_last);
-      x_fill_rectangle (f, f->output_data.x->normal_gc,
-			x0, y1 - 1, x1 - x0, 1, false);
-    }
-  else
-    {
-    /* In any other case do not draw the first and last pixels
-       differently.  */
-      XSetForeground (display, f->output_data.x->normal_gc, color);
-      x_fill_rectangle (f, f->output_data.x->normal_gc,
-			x0, y0, x1 - x0, y1 - y0, false);
-    }
+  cairo_t *cr;
+
+  cr = x_begin_cr_clip (f, f->output_data.x->normal_gc);
+  cairo_set_source_rgba(cr, 1, 1, 1, 0);
+  cairo_set_operator(cr, CAIRO_OPERATOR_SOURCE);
+  cairo_rectangle (cr, x0, y0, x1-x0, y1-y0);
+  cairo_fill (cr);
+  x_end_cr_clip (f);
 }
 
 #ifdef HAVE_XDBE
-- 
2.41.0

